design_docs/mock-tx-info.md

```diff
@@ -0,0 +1,112 @@
+# Mocking TxInfo
+
+## Context
+
+Some use-cases like testing account contracts may require mocking the contents of `TxInfo` struct returned
+by `get_tx_info()` syscall.
+
+As of writing of this document, `TxInfo` is defined as follows
+
+```cairo
+struct TxInfo {
+    // The version of the transaction. It is fixed (currently, 1) in the OS, and should be
+    // signed by the account contract.
+    // This field allows invalidating old transactions, whenever the meaning of the other
+    // transaction fields is changed (in the OS).
+    version: felt252,
+    // The account contract from which this transaction originates.
+    account_contract_address: ContractAddress,
+    // The max_fee field of the transaction.
+    max_fee: u128,
+    // The signature of the transaction.
+    signature: Span<felt252>,
+    // The hash of the transaction.
+    transaction_hash: felt252,
+    // The identifier of the chain.
+    // This field can be used to prevent replay of testnet transactions on mainnet.
+    chain_id: felt252,
+    // The transaction's nonce.
+    nonce: felt252,
+}
+```
+
+## Goal
+
+Propose a solution for mocking contexts of `TxInfo`.
+
+## Proposed solution
+
+Introduce a new cheatcode `start_mock_tx_info` (working title) with a following signature
+
+```cairo
+fn start_mock_tx_info(contract_address: felt252, tx_info: NewTxInfo) -> Result::<(), felt252> nopanic
```

Comment:
In what scenario it could fail? Imo it should just return `()`

Reply:
I don't think there's a viable scenario for this to fail, but all our cheatcodes currently are defined to return some kind of error code.

Reply:
We could revisit that for cheatcodes that cannot fail.